src/Cleave.react.js

```diff
@@ -308,7 +361,15 @@ var Cleave = React.createClass({
         return (
             <input
                 type="text"
-                ref={htmlRef}
+                ref={(ref) => {
+                    this.element = ref;
+
+                    if (!htmlRef) {
+                        return;
+                    }
+
+                    htmlRef.apply(this, arguments);
```

Comment:
I'm pretty sure this won't work.  We don't want to change the `this` of `htmlRef`.

Reply:
The last line ``htmlRef.apply(this, arguments);`` is roughly the same as we would pass ``htmlRef`` directly as it was done in the previous version, so this line should be fine.

However, the problem is with the arrow function that I've mistakenly used (which in this case binds a different value to ``this``, namely the component itself). The arrow function should be replaced with a regular anonymous function an upper scope's ``this`` capture, to use with ``this.element``.

I'm not sure if this is what you've suggested with your comment, but nevertheless thanks for pointing out there might be a problem here.